### ISSUE TITLE
`git push` of `example-dockerfile-http` fails with python backtrace

### ISSUE THREAD

**Issue by slack** [deis/dockerbuilder#21]
Push of `deis/example-dockerfile-http` application to current (2/17/2016) deis-dev manifests.

Looks like `deploy.py` exploded with backtrace:

``` console
k:beef ~ $ kd logs dockerbuild-grassy-sailfish-93bee0e5-84caaf45
download tar file complete
extracting tar file complete
Step 1 : FROM alpine:3.1
Traceback (most recent call last):
  File "/deploy.py", line 50, in <module>
    log_output(stream, True)
  File "/deploy.py", line 15, in log_output
    print(chunk.get('stream').replace('\n',''))
AttributeError: 'NoneType' object has no attribute 'replace'
```

Full logs: https://gist.github.com/slack/663eb20c5f508be13f3d

**Comment by smothiki** [comment]
This is an intermittent issue. Some times docker client bails out

**Comment by mboersma** [comment]
I fixed that particular stacktrace with #22.

 But there may still be an underlying error going on: we are examining a stream and reading from "chunks" that act like python `dict`s, so it's possible there is an "error" key or another attribute the code should also be watching for here.

Having said that, I've had good luck with example-dockerfile-http and example-dockerfile-python lately on Deis v2.

**Comment by slack** [comment]
Yeah, I think the takeaway is that the code in `deploy.py` needs to be far more defensive.

**Comment by bacongobbler** [comment]
closing as the original issue was fixed in #22 as @mboersma pointed out. If there's more bugs which we need to fix `deploy.py` then let's file another issue.
